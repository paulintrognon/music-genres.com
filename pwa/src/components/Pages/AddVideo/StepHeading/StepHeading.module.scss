.container {
  position: relative;
  max-width: 100%;
  font-family: 'Montserrat';
  font-weight: 600;
  text-transform: uppercase;
}

.backgroundText {
  font-weight: 900;
  font-size: 6.75rem;
  line-height: 5.625rem;
  color: var(--color-tertiary);
  text-align: center;
  white-space: nowrap;
  overflow: hidden;

  &.one {
    font-size: 7.25rem;
  }
  &.two {
    font-size: 7.25rem;
    letter-spacing: -0.475rem;
  }
}

.foregroundText {
  position: absolute;
  bottom: 0;
  font-family: 'Montserrat';
  font-size: 1.5rem;
  line-height: 1.5rem;
}

.stepText {
  position: absolute;
  right: 0;
  bottom: 0;
}