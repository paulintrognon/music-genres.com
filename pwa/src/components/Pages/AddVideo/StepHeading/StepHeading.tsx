import classnames from 'classnames'
import classes from './StepHeading.module.scss'

interface Props {
  step: 1 | 2
}
const StepHeading: React.FC<Props> = ({ step, children }) => {
  const stepWord = step === 1 ? 'one' : 'two'
  return (
    <div className={classes.container}>
      <div className={classnames(classes.backgroundText, classes[stepWord])}>Step {stepWord}</div>
      <div className={classes.stepText}>{step}/2</div>
      <h2 className={classes.foregroundText}>{children}</h2>
    </div>
  )
}
export default StepHeading
