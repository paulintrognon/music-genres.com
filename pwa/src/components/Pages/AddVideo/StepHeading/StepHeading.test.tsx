import React from 'react'
import { createRenderer } from 'react-test-renderer/shallow'
import StepHeading from './StepHeading'

const renderer = createRenderer()

it('should render the StepHeading (step 1)', () => {
  expect(renderer.render(<StepHeading step={1}>Super heading</StepHeading>)).toMatchSnapshot()
})

it('should render the StepHeading (step 2)', () => {
  expect(renderer.render(<StepHeading step={2}>Super heading</StepHeading>)).toMatchSnapshot()
})
