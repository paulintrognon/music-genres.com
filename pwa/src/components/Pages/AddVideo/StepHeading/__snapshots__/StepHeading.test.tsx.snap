// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the StepHeading (step 1) 1`] = `
<div
  className="container"
>
  <div
    className="backgroundText one"
  >
    Step 
    one
  </div>
  <div
    className="stepText"
  >
    1
    /2
  </div>
  <h2
    className="foregroundText"
  >
    Super heading
  </h2>
</div>
`;

exports[`should render the StepHeading (step 2) 1`] = `
<div
  className="container"
>
  <div
    className="backgroundText two"
  >
    Step 
    two
  </div>
  <div
    className="stepText"
  >
    2
    /2
  </div>
  <h2
    className="foregroundText"
  >
    Super heading
  </h2>
</div>
`;
